src/implementations/vanilla/components/select/select.js

```diff
@@ -424,11 +440,31 @@ export class Select {
     });
     // Select all checkbox follows along.
     const checked = visible.filter((c) => c.querySelector('input').checked);
-    if (visible.length !== checked.length) {
+    if (visible.length === 0 || visible.length !== checked.length) {
       this.selectAll.querySelector('input').checked = false;
     } else {
       this.selectAll.querySelector('input').checked = true;
     }
+    // Display no-results message.
+    const noResultsElement = this.searchContainer.querySelector(
+      '.ecl-select__multiple-no-results'
+    );
+    if (visible.length === 0 && noResultsElement === null) {
```

Comment:
It's a correct check  `=== null` but why not simply writing `!noResultsElement`? Up to you, as i said there is nothing wrong about this. Same happens a few line later and same approach can be used.